lib/settings/settings_editors.dart

```diff
@@ -97,20 +98,23 @@ class SettingsEditorsScreenState extends State<SettingsEditorsScreen> {
       ),
       ProOverlay(
         feature: Feature.singleJournalEntry,
-        child: ListPreference(
-          title: context.loc.settingsNoteNewNoteFileName,
-          currentOption:
-              folderConfig.journalFileNameFormat.toPublicString(context),
-          options: NoteFileNameFormat.options
-              .map((f) => f.toPublicString(context))
-              .toList(),
-          onChange: (String publicStr) {
-            var format =
-                NoteFileNameFormat.fromPublicString(context, publicStr);
+        child: NoteFileNameFormatPreference(
+          getFileNameFormatFromConfig: () => folderConfig.journalFileNameFormat,
+          setFileNameFormatInConfig: (format) {
             folderConfig.journalFileNameFormat = format;
             folderConfig.save();
             setState(() {});
```

Comment:
Not sure what this is for, but it was present in the previous `onChange` handler, so included it here and below.